web-app/src/games/games.service.ts

```diff
@@ -106,18 +136,21 @@ export class GamesService {
   ): GameAndDisplayStatus {
     const gameAndStatus: GameAndDisplayStatus = {
       game: game,
-      readyToPlay: this.haveTwoPlayersJoinedIn(game),
-      currentPlayerTurnToPlay:
+      readyToPlay: this.haveTwoPlayersJoinedIn(game) || game.soloMode,
+      currentPlayerTurnToPlay: !!(
         (game.state.turn === 'white' && playerName === game.player1.username) ||
-        (game.state.turn === 'black' && playerName === game.player2.username),
+        (game.state.turn === 'black' &&
+          game.player2 &&
+          playerName === game.player2.username)
+      ),
```

Comment:
I think it becomes harder to read it and could benefit from being extracted into it's own well named function instead, e.g isItMyTurnToPlay(askingPlayer, game). Ok it's not very well named but i don't have a better idea for now ^^